apsuite/optimization/base.py

```diff
@@ -306,3 +336,21 @@ def _target_func(self):
         except OptimizationAborted:
             _log.info('Exiting: stop event was set.')
         self._finalization()
+
+    def _get_cumulated_optimum_indices(self):
+        """Get the indices of the optima found during objfunc evaluations.
+
+        self.idcs_cumulated_optimum is an m-array with the accumulated optimum
+        indices along the `self.objfuncs_evaluated` list. Assumes
+        single-objective, single-population optimzation algorithms, i.e.
+        `self.objfuncs_evaluated` is a list of scalar values.
+        """
+        funcs = _np.array(self.objfuncs_evaluated)
+
+        optima = []
+        for i in range(len(funcs)):
+            funcsi = funcs[:i+1]
+            funcsi = funcsi[~_np.isnan(funcsi)]
+            optima.append(funcsi.min())
+
+        self.idcs_cumulated_optimum = _np.isin(funcs, optima).nonzero()[0]
```

Comment:
I think there are two problems here:
 1 - This method is not general enough to handle multi-objective and multi-population algorithms;
 2 - In the case of single-objective functions, I think this algorithm over-complicate the task and it also looks to be very slow. Have you considered the simple approach below:
```suggestion
        optima = []
        mini = _np.inf
        for i, fun in enumerate(funcs):
            if not _np.isnan(fun) and fun < mini:
                optima.append(i)
                mini = fun
        self.idcs_cumulated_optimum = _np.array(optima)
```

Reply:
Thanks for the suggestion, Fernando. Indeed it was too complicated. I have implemented your suggestion and also generalized it to the search of cumulative optima along a certain objective and certain individual for multi-objective and multi-individual algorithms. I also generalized the `plot_knobspace_slice`.